chartmogul/retry_request.py

```diff
@@ -0,0 +1,25 @@
+import requests
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+METHOD_WHITELIST = ['HEAD', 'GET', 'POST', 'PUT', 'DELETE', 'OPTIONS', 'TRACE']
+STATUS_FORCELIST = (429, 500, 502, 503, 504, 520, 524)
+
+def requests_retry_session(retries=20, backoff_factor=2, session=None,):
+    session = session or requests.Session()
+    adapter = _retry_adapter(retries, backoff_factor)
+    session.mount('http://', adapter)
```

Comment:
Will this allow requests to be made over HTTP? If so, do you think we should remove it?

Reply:
Removed.